File: model/preferences.go
// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See License.txt for license information.

package model

import (
	"encoding/json"
	"io"

	"github.com/json-iterator/go"
)

type Preferences []Preference

func (o *Preferences) ToJson() string {
	b, _ := jsoniter.Marshal(o)
	return string(b)
}

func PreferencesFromJson(data io.Reader) (Preferences, error) {
	decoder := json.NewDecoder(data)
	var o Preferences
	err := decoder.Decode(&o)
	if err == nil {
		return o, nil
	} else {
		return nil, err
	}
}

File: webapp/platform/types/src/preferences.ts
// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export type PreferenceType = {
    category: string;
    name: string;
    user_id: string;
    value: string;
};

export type PreferencesType = {
    [x: string]: PreferenceType;
};
